lib/eco/treemanager.py

```diff
@@ -628,7 +629,14 @@ def jump_to_error(self, parser):
 
     def key_shift_ctrl_z(self):
         self.log_input("key_shift_ctrl_z")
-        if self.get_max_version() > self.version:
+        try:
+            i = self.undo_snapshots.index(self.version)
+            if i == len(self.undo_snapshots) - 1:
+                return
+            undo_amount = self.undo_snapshots[i+1] - self.undo_snapshots[i]
+        except ValueError:
```

Comment:
The except returns, so it doesn't matter. But I'm happy to just leave it as it is.

I think it's ready to merge.